### PR TITLE
[Plugin] adding starship

### LINKED ISSUE #627: [Plugin] Support to Starship
https://starship.rs/

### PR DESCRIPTION
**Pull request by Heus-Sueh**
close my own issue: https://github.com/ohmybash/oh-my-bash/issues/627

I believe I didn't do anything wrong in the code

----

Closes #627

### CODE REVIEW

**Comment by akinomyoga** [comment]
`starship` conflicts with `OSH_THEME` since both provide `PS1`. Shouldn't this be implemented as a theme?

**Comment by akinomyoga** [comment]
In addition, our themes are supposed to be able to switch to another theme when requested, i.e. a theme can be unloaded and another theme can be loaded. Starship can be enabled by `eval "$(starship init bash)"`, but is there a way to unload Starship in a Bash session where Starship was loaded?

**Comment by Heus-Sueh** [comment]
> `starship` conflicts with a theme specified by `OSH_THEME` since both provide `PS1`. Shouldn't this be implemented as a theme?

I based it on what [ohmyzsh did](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/starship), which is a plugin instead of a conventional theme

> but is there a way to unload Starship in a Bash session where Starship was loaded?

Looking at the starship commands, I didn't find a way to turn it off in an active session


maybe I should add this giant warning:

`⚠️ ENABLING THIS PLUGIN WILL UNSET YOUR OSH_THEME VARIABLE`

?

File: plugins/starship/starship.plugin.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! bash oh-my-bash.module
+
+# Check if starship is installed
+if _omb_util_command_exists starship; then
+    # ignore oh-my-bash theme
+    unset OSH_THEME
```
Comment (akinomyoga) [inline plugins/starship/starship.plugin.sh:6]:
```suggestion
    unset -v OSH_THEME
```

**Review by akinomyoga — APPROVED** [review]

**Comment by akinomyoga** [comment]
Although providing an external theme through a "plugin" instead of a "theme" seems wrong, it should probably better for the consistency with the upstream Oh My Zsh.

**Pull request merged**